File: exercises/02 stubs and mocks/java/src/eu/sioux/tdd/stubsmocks/Notifier.java
package eu.sioux.tdd.stubsmocks;

import java.util.ArrayList;
import java.util.List;

public class Notifier {
    private List<ResponseHandler> responseHandlers = new ArrayList<>();

    public void add(ResponseHandler subscriber) {
        responseHandlers.add(subscriber);
    }

    public void publish(String message) {
        for (ResponseHandler responseHandler : responseHandlers) {
            System.out.println("Notifying " + responseHandler.getName() + " with message: " + message);
            responseHandler.receive(message);
        }
    }
}

File: examples/02 stubs and mocks/cs/StubsMocks/Notifier.cs
using System;
using System.Collections.Generic;

namespace StubsMocks
{
    class Notifier
    {
        private IList<IResponseHandler> responseHandlers = new List<IResponseHandler>();

        public void Add(IResponseHandler subscriber)
        {
            responseHandlers.Add(subscriber);
        }

        public void Publish(string message)
        {
            foreach (var handler in responseHandlers)
            {
                Console.WriteLine($"Notifying {handler.Name} with message: {message}");
                handler.Receive(message);
            }
        }
    }
}
